bundles/persistence/org.openhab.persistence.influxdb/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -415,24 +421,31 @@ private Object convertBigDecimalToNum(BigDecimal value) {
   /**
    * Converts {@link State} to objects fitting into influxdb values.
    * 
-   * @param state to be converted
-   * @return integer or double value for DecimalType and PercentType, an integer for DateTimeType
-   *         and 0 or 1 for OnOffType and OpenClosedType.
+   * @param 	state to be converted
+   * @return 	integer or double value for DecimalType and PercentType, 
+   * 			0 or 1 for OnOffType and OpenClosedType,
+   *         	SortedMap<String, PrimitiveType> for HSBType,
+   *         	integer for DateTimeType,
+   *         	SortedMap<String, PrimitiveType> for PointType
    */
   private Object stateToObject(State state) {
     Object value;
-    if (state instanceof PercentType) {
-      value = convertBigDecimalToNum(((PercentType) state).toBigDecimal());
-    } else if (state instanceof DecimalType) {
-      value = convertBigDecimalToNum(((DecimalType) state).toBigDecimal());
-    } else if (state instanceof DateTimeType) {
-      value = ((DateTimeType) state).getCalendar().getTime().getTime();
+    if (state instanceof DecimalType) {
+    	value = convertBigDecimalToNum(((DecimalType) state).toBigDecimal());
+    } else if (state instanceof PercentType) {
```

Comment:
Remove handling of PercentType because it is a subtype of DecimalType and both are converted in the same way.